backend/src/main/java/com/team4/backend/controller/FileMetaDataController.java

```diff
@@ -26,10 +26,13 @@
 
     @PostMapping
     @PreAuthorize("hasAuthority('STUDENT')")
-    public Mono<ResponseEntity<String>> uploadFile(@RequestPart("filename") String filename, @RequestPart("type") String type, @RequestPart("mimeType") String mimeType, @RequestPart("file") Mono<FilePart> filePartMono, Principal principal) {
-        return fileMetaDataService.uploadFile(filename, type, mimeType, filePartMono, UserSessionService.getLoggedUserEmail(principal))
-                .flatMap(u -> Mono.just(ResponseEntity.status(HttpStatus.CREATED).body("")))
-                .onErrorResume(error -> Mono.just(ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(error.getMessage())));
+    public Mono<ResponseEntity<String>> uploadFile(@RequestPart("filename") String filename,
+            @RequestPart("type") String type, @RequestPart("mimeType") String mimeType,
+            @RequestPart("file") Mono<FilePart> filePartMono, Principal principal) {
+        return fileMetaDataService
+                .uploadFile(filename, type, mimeType, filePartMono, UserSessionService.getLoggedUserEmail(principal))
+                .flatMap(u -> Mono.just(ResponseEntity.status(HttpStatus.CREATED).body(""))).onErrorResume(error -> Mono
+                        .just(ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(error.getMessage())));
```

Comment:
No error is thrown manually in the service, this onErrorResume is here to catch said exception so that the application mais continue to run even if an error happens.

Reply:
the application still runs if its a verified exception (not a runtime error)
and it will be a 500 error

Reply:
👍